plugins/destination/clickhouse/typeconv/arrow/values/time.go

```diff
@@ -48,16 +70,5 @@ func timeToTimestamp(value time.Time, tsType *arrow.TimestampType) (arrow.Timest
 		value = value.In(loc)
 	}
 
-	switch tsType.Unit {
-	case arrow.Second:
-		return arrow.Timestamp(value.Unix()), nil
-	case arrow.Millisecond:
-		return arrow.Timestamp(value.Unix()*1e3 + int64(value.Nanosecond())/1e6), nil
-	case arrow.Microsecond:
-		return arrow.Timestamp(value.Unix()*1e6 + int64(value.Nanosecond())/1e3), nil
-	case arrow.Nanosecond:
-		return arrow.Timestamp(value.UnixNano()), nil
-	default:
-		return arrow.Timestamp(0), fmt.Errorf("unsupported Apache Arrow time unit: %s", tsType.Unit.String())
-	}
+	return arrow.TimestampFromTime(value, tsType.Unit)
```

Comment:
To confirm, this is a refactoring unrelated to the fixes needed for the upgrade?